dao/chat.go

```diff
@@ -2,8 +2,48 @@ package dao
 
 import (
 	"TUM-Live/model"
+	"errors"
+	"github.com/go-sql-driver/mysql"
 )
 
 func AddMessage(chat *model.Chat) error {
 	return DB.Save(chat).Error
 }
+
+//ToggleLike adds a like to a message from the user if it doesn't exist, or removes it if it does
+func ToggleLike(userID uint, chatID uint) error {
+	err := DB.Exec("INSERT INTO chat_user_likes (user_id, chat_id) VALUES (?, ?)", userID, chatID).Error
+	if err == nil {
+		return nil // like was added successfully
+	}
+	var mysqlErr *mysql.MySQLError
+	if errors.As(err, &mysqlErr) && mysqlErr.Number == 1062 { // 1062: duplicate entry -> message already liked -> remove
+		return DB.Exec("DELETE FROM chat_user_likes WHERE user_id = ? AND chat_id = ?", userID, chatID).Error
+	}
+	return err // some other error
+}
+
+// GetNumLikes returns the number of likes for a message
+func GetNumLikes(chatID uint) (int, error) {
+	var numLikes int
+	err := DB.Raw("SELECT COUNT(*) FROM chat_user_likes WHERE chat_id = ?", chatID).Scan(&numLikes).Error
+	return numLikes, err
+}
+
+// GetChats returns all chats for the stream with the given ID. Number of likes are inserted and the user's like status is determined
+func GetChats(userID uint, streamID uint) ([]model.Chat, error) {
+	var chats []model.Chat
+	err := DB.Preload("Replies").Preload("UserLikes").Find(&chats, "stream_id = ?", streamID).Error
+	if err == nil {
+		for i := range chats {
```

Comment:
```suggestion
		for _, chat := range chats {
```
Matter of taste, but I think it is safer to omit the index. Rest should be updated to the new variable.

Reply:
Since I'm assigning here and the chat is not a pointer, I'd like to keep the index here.